Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1148,74 +1154,74 @@ namespace AZ
 
-                if (!reflectionProbes.empty() && reflectionProbes[0])
-                {
-                    m_shaderResourceGroup->SetConstant(modelToWorldConstantIndex, reflectionProbes[0]->GetTransform());
-                    m_shaderResourceGroup->SetConstant(modelToWorldInverseConstantIndex, Matrix3x4::CreateFromTransform(reflectionProbes[0]->GetTransform()).GetInverseFull());
-                    m_shaderResourceGroup->SetConstant(outerObbHalfLengthsConstantIndex, reflectionProbes[0]->GetOuterObbWs().GetHalfLengths());
-                    m_shaderResourceGroup->SetConstant(innerObbHalfLengthsConstantIndex, reflectionProbes[0]->GetInnerObbWs().GetHalfLengths());
-                    m_shaderResourceGroup->SetConstant(useReflectionProbeConstantIndex, true);
-                    m_shaderResourceGroup->SetConstant(useParallaxCorrectionConstantIndex, reflectionProbes[0]->GetUseParallaxCorrection());
-                    m_shaderResourceGroup->SetConstant(exposureConstantIndex, reflectionProbes[0]->GetRenderExposure());
-
-                    m_shaderResourceGroup->SetImage(reflectionCubeMapImageIndex, reflectionProbes[0]->GetCubeMapImage());
-                }
-                else
-                {
-                    m_shaderResourceGroup->SetConstant(useReflectionProbeConstantIndex, false);
+                    if (!reflectionProbes.empty() && reflectionProbes[0])
+                    {
+                        objectSrg->SetConstant(modelToWorldConstantIndex, reflectionProbes[0]->GetTransform());
+                        objectSrg->SetConstant(modelToWorldInverseConstantIndex, Matrix3x4::CreateFromTransform(reflectionProbes[0]->GetTransform()).GetInverseFull());
+                        objectSrg->SetConstant(outerObbHalfLengthsConstantIndex, reflectionProbes[0]->GetOuterObbWs().GetHalfLengths());
+                        objectSrg->SetConstant(innerObbHalfLengthsConstantIndex, reflectionProbes[0]->GetInnerObbWs().GetHalfLengths());
+                        objectSrg->SetConstant(useReflectionProbeConstantIndex, true);
+                        objectSrg->SetConstant(useParallaxCorrectionConstantIndex, reflectionProbes[0]->GetUseParallaxCorrection());
+                        objectSrg->SetConstant(exposureConstantIndex, reflectionProbes[0]->GetRenderExposure());
+
+                        objectSrg->SetImage(reflectionCubeMapImageIndex, reflectionProbes[0]->GetCubeMapImage());
+                    }
+                    else
+                    {
+                        objectSrg->SetConstant(useReflectionProbeConstantIndex, false);
+                    }
                 }
+
+                objectSrg->Compile();
             }
 
-            m_shaderResourceGroup->Compile();
-            m_objectSrgNeedsUpdate = false;
+            // Set m_objectSrgNeedsUpdate to false if there are object SRGs in the list
+            m_objectSrgNeedsUpdate = m_objectSrgNeedsUpdate && (m_objectSrgList.size() == 0);
```

Comment:
This logic is a kind of hard to wrap my head around. (Maybe a personal problem :) )
Can't we just set the flag to false like before?

Reply:
This is exactly the same logic as
if(m_objectSrgList.size() > 0)
    m_objectSrgNeedsUpdate = false;
This is to mirror the previous logic where we earlied out if m_shaderResourceGroup was null.